### PR TITLE
Add new rendering settings to `LabelSettings` and in `Label` theme

### LINKED ISSUE #10383: Add to Label new rendering settings in theme and LabelSettings
### Describe the project you are working on

2d action runner

### Describe the problem or limitation you are having in your project

You cannot set the outline color of the shadow, although you can set the thickness of the shadow's stroke. You can't change the rendering sequence of the shadow so that it would be like in the 3 versions of the engine (the shadow would be between the text and the text stroke). Sometimes you need to create the effect of convex or voluminous text that is possible only when duplicating Label with different colors and each time synchronize their value text that is very undesirable or shader that is undesirable on mobile devices 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Adds to Label Theme and LabelSettings advanced text rendering settings and the ability to change the order in which shadows are rendered. Finalize the rendering of the shadow stroke so that it is generated in a separate step. Implement the ability to set the color for the shadow stroke, not duplicate the shadow color. Realize the ability to change the order of shadow rendering to create the same effect of rendering shadows as was godot 3-x. This will help to better customize the visual for label.

-An example of a required result that cannot be done with the current Label render settings and without using additional nodes and shaders
image

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

[PR](https://github.com/godotengine/godot/pull/95214)

Theme Overrides for Label and LabelSettings include 2 new parameters: font_shadow_outline_color and draw_shadow before outline. Now for the shadow, the shadow outline is still drawn, you just need to draw the shadow outline in a separate step to avoid overlap and set the color through a separate variable, and not use the same color as the shadow

To realize different postdates of drawing queues, before executing the loop we will form an array of drawing steps and go through the contents of the array

Label.cpp:
```
LabelDrawStep draw_steps[DRAW_STEP_MAX];

if (shadow_before_outline) {
	draw_steps[0] = DRAW_STEP_OUTLINE;
	draw_steps[1] = DRAW_STEP_SHADOW_OUTLINE;
	draw_steps[2] = DRAW_STEP_SHADOW;
	draw_steps[3] = DRAW_STEP_TEXT;
} else {
	draw_steps[0] = DRAW_STEP_SHADOW_OUTLINE;
	draw_steps[1] = DRAW_STEP_SHADOW;
	draw_steps[2] = DRAW_STEP_OUTLINE;
	draw_steps[3] = DRAW_STEP_TEXT;
}

// Draw shadow outline, shadow, outline and text. Note: Do not merge this into the single loop iteration, to prevent overlaps.
for (int step : draw_steps) {
```

Examle Theme
bn_1

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

These changes affect the Label rendering logic in `Label.cpp`, so this should be core

### PR DESCRIPTION
**Pull request by xolarkodak**
This PR adds to `Label` Theme and `LabelSettings `advanced text rendering settings and the ability to change the order in which shadows are rendered.
I think this PR is quite important in terms of visual label display, as it eliminates the need in some cases to use shaders that are costly on weak or medium mobile devices. This PR does not affect optimization in any way, but only increases functionality and usability.  The purpose of this PR is to add a toolkit for the ability to create even more variant and colorful fonts out of the box. The changes that have been made are not coordinate  and are therefore stable. It is important to note that it was possible to create an alternative to this function using 2 labels with the same text but different colors, which is very uncomfortable and less optimized than this implementation and moreover thanks to the implementation within the engine you can easily and conveniently use pre-prepared themes for text styling.

**Further changes to this PR include:**
- corrections in shadow outline rendering
- implementation of functionality for adjusting the shadow outline color
- the ability to change the order in which the shadow is rendered.
- the editor visually changed the order of constants and colors in the theme
- changed `shadow_outline_size `parameter in `default_theme.cpp`: `1` -> `0`

Overview of changes:
- theme overrides
image

- shadow outline color
https://github.com/user-attachments/assets/bd7df539-1981-445c-a689-821fc9f0069d

- draw shadow before text outline
https://github.com/user-attachments/assets/f204a5c3-aae3-4422-a8fc-c8a41b715ea9

- a clear example of the difference `draw_shadow_before_outline `mode
https://github.com/user-attachments/assets/c1dd7991-5b11-4373-9721-ddca7fc834e2

Results using this functional:
image
image

Built version of the engine (from Godot 4.3-rc2) with implementation of the demonstrated functionality:
- Build parameters: `scons platform=windows dev_mode=yes tests=yes`
[Google drive](https://drive.google.com/file/d/1P9wQGT9-uqWvfBoJmmnd0DveBlwmenRb/view?usp=sharing)

* *Bugsquad edit, closes: https://github.com/godotengine/godot-proposals/issues/10383*

### CODE REVIEW

File: doc/classes/Label.xml [hunk 1]
```diff
@@ -129,14 +132,20 @@
...
 		<theme_item name="shadow_offset_x" data_type="constant" type="int" default="1">
 			The horizontal offset of the text's shadow.
 		</theme_item>
 		<theme_item name="shadow_offset_y" data_type="constant" type="int" default="1">
 			The vertical offset of the text's shadow.
 		</theme_item>
-		<theme_item name="shadow_outline_size" data_type="constant" type="int" default="1">
-			The size of the shadow outline.
+		<theme_item name="draw_shadow_before_outline" data_type="constant" type="int" default="0">
+			Changes the order in which the [code]shadow[/code] and [code]shadow_outline[/code] is draw. If the value is [code]1[/code], the [code]shadow[/code] and the [code]shadow_outline[/code] will be [b]between[/b] the [code]text_outline[/code] and the main [code]text[/code], otherwise the shadow will be in the background by default
```
Comment (Calinou) [inline doc/classes/Label.xml:145]:
```suggestion
			Changes the order in which the font [code]shadow[/code] and [code]shadow_outline[/code] is drawn. If the value is [code]1[/code], the [code]shadow[/code] and the [code]shadow_outline[/code] will be [b]between[/b] the [code]text_outline[/code] and the main [code]text[/code], otherwise the shadow will be in the background by default.
```

**Comment by Calinou** [comment]
Thanks for opening a pull request :slightly_smiling_face: 

Feature pull requests should be associated to a *feature proposal* to justify the need for implementing the feature in Godot core. Please open a proposal on the [Godot proposals](https://github.com/godotengine/godot-proposals) repository (and link to this pull request in the proposal's body).

File: scene/resources/label_settings.cpp [hunk 2]
```diff
@@ -74,9 +82,11 @@ void LabelSettings::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::COLOR, "outline_color"), "set_outline_color", "get_outline_color");
 
 	ADD_GROUP("Shadow", "shadow_");
-	ADD_PROPERTY(PropertyInfo(Variant::INT, "shadow_size", PROPERTY_HINT_RANGE, "0,127,1,or_greater,suffix:px"), "set_shadow_size", "get_shadow_size");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "shadow_outline_size", PROPERTY_HINT_RANGE, "0,127,1,or_greater,suffix:px"), "set_shadow_outline_size", "get_shadow_outline_size");
```
Comment (AThousandShips) [inline scene/resources/label_settings.cpp:85]: This correction breaks compatibility, so I don't think this correction is appropriate

Comment (Mickeon) [reply scene/resources/label_settings.cpp:83]: In layman's terms, the existing property should still exist but be deprecated.

See other deprecated properties for a bit of inspiration (for example Control's `auto_translate`)

**Review by bruvzg — COMMENTED** [review]
Current Label shadow is a bit hacky instead of adding more narrow use properties on top of it, it might be better to make it more universal (it will be breaking change anyway):

Something like an arbitrary size array of draw steps, each defining `color`, `offset`, and `outline size`.

**Comment by xolarkodak** [comment]
> Current Label shadow is a bit hacky instead of adding more narrow use properties on top of it, it might be better to make it more universal (it will be breaking change anyway):
> 
> Something like an arbitrary size array of draw steps, each defining `color`, `offset`, and `outline size`.

That's a very interesting idea. Could you please describe it in more detail. So you want to make it possible to implement an arbitrary number of shadows and each shadow will have an individual setting of offset color and stroke size? I still have little experience I think I have architecturally correct will not be able to realize such a realization.
